app/models/user.rb

```diff
@@ -0,0 +1,25 @@
+require 'bcrypt'
+
+class User
+
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :username, String, unique: true
+  property :name, String
+  property :email, String, unique: true
+  property :password_digest, Text
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  validates_confirmation_of :password
+  validates_uniqueness_of :username
+  validates_uniqueness_of :email
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.